packages/server-setup/tasks-handlers.ts

```diff
@@ -67,40 +75,38 @@ const handlers = {
       return;
     }
 
-    const { stdout, stderr, exitCode } =
-      await $`hapi-fhir-cli upload-terminology -d "${TERMINOLOGIES_DATA_BASEPATH}${dataSource}" -v "${dataVersion}" -t "${server.url}" -u "${dataType}"`;
-    logResults(stdout, stderr, exitCode);
-
-    createApplicationImportAuditEvent(
-      server.url,
-      {
-        source: dataSource,
-        system: dataType,
-        version: dataVersion,
-      },
-      exitCode === 0 ? "0" : "8"
-    );
+    try {
+      if (plugin) {
+        await plugin.uploadTerminology(server, task);
+      } else {
+        const { stdout, stderr, exitCode } =
+          await $`hapi-fhir-cli upload-terminology -d "${TERMINOLOGIES_DATA_BASEPATH}${dataSource}" -v "${dataVersion}" -t "${server.url}" -u "${dataType}"`;
+        logResults(stdout, stderr, exitCode);
+      }
+    } catch (e) {
+      console.error(e);
+      outcome = "8";
+    } finally {
+      await createApplicationImportAuditEvent(
+        server.url,
+        {
+          source: dataSource,
+          system: dataType,
+          version: dataVersion,
+        },
+        outcome
+      );
+    }
```

Comment:
this try/catch doesn't cover the case where
- a task config had a `plugin` property
- the plugin is not found

Reply:
✅